psycop/common/sequence_models/tasks/tasks.py

```diff
@@ -167,6 +170,20 @@ def collate_fn(
         batch_with_labels = self.masking_fn(padded_sequence_ids)
         return batch_with_labels
 
+    def on_load_checkpoint(self, checkpoint: dict[str, torch.Tensor]) -> None:
+        self.mlm_head = nn.Linear(self.d_model, self.embedder.n_diagnosis_codes)
+        state_dict = checkpoint["state_dict"]
+
+        mlm_head_state_dict = {
+            "weight": state_dict["mlm_head.weight"],  # type ignore
+            "bias": state_dict["mlm_head.bias"],  # type: ignore
+        }
+        self.mlm_head.load_state_dict(mlm_head_state_dict)
```

Comment:
This seems very dirty, but since the mlm_head is not initialised in init, it is not correctly initialised when loading from checkpoint otherwise.

I considered using setup, but that re-initialised on every fit, predict etc., which I think means we'll lose the weights and biases of the mlm head.

Reply:
Hooks are documented here: https://lightning.ai/docs/pytorch/stable/common/lightning_module.html#hooks